src/getters/awaitGraphBlock.ts

```diff
@@ -0,0 +1,32 @@
+import getGraphClient from '~/getters/getGraphClient'
+import {
+    GetMetadataDocument,
+    GetMetadataQuery,
+    GetMetadataQueryVariables,
+} from '~/generated/gql/network'
+
+export const awaitGraphBlock = (blockNumber: number): Promise<void> => {
+    return new Promise((resolve) => {
+        const interval = setInterval(async () => {
+            const client = getGraphClient()
+            const {
+                data: { _meta: meta },
+            } = await client.query<GetMetadataQuery, GetMetadataQueryVariables>({
+                query: GetMetadataDocument,
+            })
+
+            if (!!meta && meta.block.number >= blockNumber) {
+                clearInterval(interval)
+                resolve()
+            }
+
+            await client.clearStore()
+        }, 500)
```

Comment:
What if the code executed within that timeout takes more than 500ms? We storm the graph with request not knowing if the previous one succeeded or failed. Is that correct?

Reply:
Something like this makes it call the remote endpoint reasonably often:

```ts
export async function awaitGraphBlock(
    blockNumber: number,
    timeoutAfterMillis = 15000,
): Promise<void> {
    let done = false

    async function fn() {
        while (true) {
            if (done) {
                return
            }

            let client: ReturnType<typeof getGraphClient> | undefined

            try {
                client = getGraphClient()

                const {
                    data: { _meta: meta },
                } = await client.query<GetMetadataQuery, GetMetadataQueryVariables>({
                    query: GetMetadataDocument,
                })

                if (meta && meta.block.number >= blockNumber) {
                    return
                }
            } finally {
                await client?.clearStore()
            }

            await new Promise((resolve) => void setTimeout(resolve, 500))
        }
    }

    try {
        await Promise.race([
            new Promise((_, reject) => {
                setTimeout(() => void reject(new Error('Timeout')), timeoutAfterMillis)
            }),
            fn(),
        ])
    } finally {
        done = true
    }
}
```

Haven't tested it but it conveys the gist.

Reply:
Ok it seems to be reasonable, I'll try it! :)